pyfstat/core.py

```diff
@@ -1004,186 +1037,189 @@ def calculate_twoF_cumulative(
+            if outside those: auto-truncated
+        num_segments_(CFS|PFS) : int
+            Number of time segments to (compute|predict) twoF.
+        custom_ax_kwargs : dict
+            Optional axis formatting options.
         savefig : bool
-            If true, save the figure in outdir
-        title, plt_label: str
-            Figure title and label
+            If true, save the figure in outdir and return taus, twoFs.
+            If false, return axes object.
+        label: str
+            Output filename (ignored unless savefig is True).
+        outdir: str
+            Output folder (ignored unless savefig is True).
 
         Returns
         -------
-        tauS, tauF : ndarray shape (N,)
-            If savefig, the times and twoF (cumulative) values
         ax : matplotlib.axes._subplots_AxesSubplot, optional
-            If savefig is False
-
+            The axes object containing the plot.
         """
-        if ax is None:
-            fig, ax = plt.subplots()
-        if injectSources:
-            pfs_input = dict(
-                h0=injectSources["h0"],
-                cosi=injectSources["cosi"],
-                psi=injectSources["psi"],
-                Alpha=injectSources["Alpha"],
-                Delta=injectSources["Delta"],
-                Freq=injectSources["fkdot"][0],
-            )
-        else:
-            pfs_input = None
-
-        taus, twoFs = self.calculate_twoF_cumulative(**kwargs)
-        ax.plot(taus / 86400.0, twoFs, label=plt_label, color=c)
-        if len(self.detector_names) > 1:
-            detector_names = self.detector_names
-            detectors = self.detectors
-            for d in self.detector_names:
-                self.detectors = d
-                self.init_computefstatistic()
-                taus, twoFs = self.calculate_twoF_cumulative(**kwargs)
-                ax.plot(
-                    taus / 86400.0,
-                    twoFs,
-                    label="{}".format(d),
-                    color=detector_colors[d.lower()],
-                )
-            self.detectors = detectors
-            self.detector_names = detector_names
 
-        if add_pfs:
-            times, pfs, pfs_sigma = self._calculate_predict_fstat_cumulative(
-                N=N, label=label, outdir=outdir, pfs_input=pfs_input
+        # Compute cumulative twoF
+        actual_tstart_CFS, taus_CFS, twoFs = self.calculate_twoF_cumulative(
+            tstart=tstart,
+            tend=tend,
+            num_segments=num_segments_CFS,
+            **CFS_input,
+        )
+        taus_CFS_days = taus_CFS / 86400.0
+
+        # Set up plot-realted objects
+        axis_kwargs = {
+            "xlabel": r"Days from $t_{{\rm start}}={:.0f}$".format(actual_tstart_CFS),
+            "ylabel": r"$\log_{10}(\mathrm{BSGL})_{\rm cumulative}$"
+            if self.BSGL
+            else r"$\widetilde{2\mathcal{F}}_{\rm cumulative}$",
```

Comment:
while we're rewriting all these blocks anyway, let's make strings more consistent:
```suggestion
            "xlabel": f"Days from $t_\\mathrm{{start}}={actual_tstart_CFS:.0f}$",
            "ylabel": "$\\log_{10}(\\mathrm{BSGL})_\\mathrm{cumulative}$"
            if self.BSGL
            else "$\\widetilde{2\\mathcal{F}}_\\mathrm{cumulative}$",
```
(may need a round of debugging)

Reply:
AFAIK, you can use `f` and raw strings at once.

I'll try this one locally.

Reply:
Yes, but the raw-string hack for latex is ugly and was leading to weird issues in some places (flake8 sometimes dislikes it too), so I started replacing it with regular strings and double escapes where possible.

Reply:
Done in 25385a56453853ae8283eddc693331407750c752